jar-submission-test/src/main/java/com/hazelcast/jet/tests/jarsubmission/JarSubmissionTest.java

```diff
@@ -0,0 +1,310 @@
+
+    private String smallJarPath;
+    private String largeJarPath;
+    private int pauseBetweenSmallJarJobs;
+    private int pauseBetweenLargeJarJobs;
+
+    public static void main(String[] args) throws Exception {
+        new JarSubmissionTest().run(args);
+    }
+
+    @Override
+    protected void init(HazelcastInstance client) throws Exception {
+        smallJarPath = property("smallJarPath", SMALL_JAR_PATH_DEFAULT);
+        largeJarPath = property("largeJarPath", LARGE_JAR_PATH_DEFAULT);
+        pauseBetweenSmallJarJobs = propertyInt("pauseBetweenSmallJarJobs", PAUSE_BETWEEN_SMALL_JAR_JOBS);
+        pauseBetweenLargeJarJobs = propertyInt("pauseBetweenLargeJarJobs", PAUSE_BETWEEN_LARGE_JAR_JOBS);
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, String name) throws Throwable {
+        Throwable[] exceptions = new Throwable[4];
+        ExecutorService executorService = Executors.newFixedThreadPool(4);
+        executorService.execute(() -> {
+            try {
+                testBatchSmallJar(client);
+            } catch (Throwable t) {
+                logger.severe("Exception in small batch jar.", t);
+                exceptions[0] = t;
+            }
+        });
+        executorService.execute(() -> {
+            try {
+                testBatchLargeJar(client);
+            } catch (Throwable t) {
+                logger.severe("Exception in large batch jar.", t);
+                exceptions[1] = t;
+            }
+        });
+        executorService.execute(() -> {
+            try {
+                testStreamSmallJar(client);
+            } catch (Throwable t) {
+                logger.severe("Exception in small stream jar.", t);
+                exceptions[2] = t;
+            }
+        });
+        executorService.execute(() -> {
+            try {
+                testStreamLargeJar(client);
+            } catch (Throwable t) {
+                logger.severe("Exception in small stream jar.", t);
+                exceptions[3] = t;
+            }
+        });
+        executorService.shutdown();
+        executorService.awaitTermination((long) (durationInMillis + DELAY_AFTER_TEST_FINISHED), MILLISECONDS);
+
+        if (exceptions[0] != null) {
+            logger.severe("Exception in small batch jar.", exceptions[0]);
+        }
+        if (exceptions[1] != null) {
+            logger.severe("Exception in large batch jar.", exceptions[1]);
+        }
+        if (exceptions[2] != null) {
+            logger.severe("Exception in small stream jar.", exceptions[2]);
+        }
+        if (exceptions[3] != null) {
+            logger.severe("Exception in large stream jar.", exceptions[3]);
+        }
+
+        for (Throwable exception : exceptions) {
+            if (exception != null) {
+                throw exception;
+            }
+        }
+    }
+
+    @Override
+    protected void teardown(Throwable t) throws Exception {
```

Comment:
Make teardown() in  AbstractSoakTest empty method.  So that you don't need to override it every time